src/registerEditorEventHandler.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { uniqueName } from './utils';
+
+import {
+	CKEditorEventHandler,
+	CKEditorEventPayload,
+	CKEditorRegisterEventArgs
+} from './types';
+
+/**
+ * Registers editor event. Allows to toggle debugging mode.
+ *
+ * @param editor instance of editor
+ * @param debug toggles debugger
+ */
+function registerEditorEventHandler( {
+	debug,
+	editor,
+	evtName,
+	handler,
+	listenerData,
+	priority
+}: CKEditorRegisterEventArgs ) {
+	const handlerId = debug && uniqueName();
+
+	let _handler: CKEditorEventHandler | undefined | null;
+
+	if ( handler ) {
```

Comment:
Can't the function return early if `handler` is not provided?

Reply:
I've changed this function's interface a little so that `handler` must be passed. It doesn't really make sense to register a handler without, well, a handler ;-).